### PR TITLE
Fix upload JSON dataset

### PR DESCRIPTION
**Pull request by maximi4321**
# Summary

This PR solves issue #97

## Type of change

- Bug fix.

## Changes

- Added `name` parameter to `jsondataloader.json`
- Corrected the description of the `name` parameter in `csvdataloader.json`, so it uses text instead of html.

## How to Test

1. Go to `DashAI/DashAI/front` and run `yarn build`
2. Go back to `DashAI/` and run `python -c "import DashAI;DashAI.run()"`
3. Navigate to the datasets tab
4. Click "New Dataset"
5. Select "TabularClassificationTask" as your task
6. Select "JSONDataloader" as your dataloader
7. Upload `DashAI/back/example_datasets/irisDataset.json` as your dataset
8. Give it a name or optionally leave blank
9. Click "Save" and the dataset should be uploaded succesfully

## Screenshots

[jsontest.webm](https://github.com/DashAISoftware/DashAI/assets/53657198/d3374bda-f8d0-407e-9eaa-1d54093104a3)

### CODE REVIEW

File: DashAI/back/dataloaders/params_schemas/csvdataloader.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
       "oneOf": [
         {
           "error_msg": "",
-          "description": "<p style='font-size: 15px;'>Dataset name</p><p style='font-size: 12px;'>If left <b>blank</b>, the name of the uploaded file will be used.<p/><a style='color: #ADD8E6;' href='https://github.com/DashAISoftware/DashAI'>More info.</a>",
+          "description": "Custom name to register your dataset, if left blank, the name of the uploaded file will be used.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/csvdataloader.json]: "Custom name to register your dataset. If no name is specified, the name of the uploaded file will be used",

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 2]
```diff
@@ -1,118 +1,129 @@
...
+  "additionalProperties": true,
+  "error_msg": "You must specify the data configuration parameters.",
+  "description": "Upload a .json file with your dataset or a .zip file with JSON files to be uploaded.",
+  "display": "modal",
+  "properties": {
+    "name": {
+      "oneOf": [
+        {
+          "error_msg": "",
+          "description": "Custom name to register your dataset, if left blank, the name of the uploaded file will be used.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]:
```suggestion
          "description": "Custom name to register your dataset. If no name is specified, the name of the uploaded file will be used",,
```

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 3]
```diff
@@ -1,118 +1,129 @@
...
+          "default": "",
+          "optional": null
+        }
+      ]
+    },
+    "data_key": {
+      "oneOf": [
+        {
+          "error_msg": "Write the name of the key that constains the data.",
+          "description": "The data has to be inside a key in your JSON files. Write the name of the key that constains the data.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]: "Name of key that contains the data in the JSON files."

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 4]
```diff
@@ -1,118 +1,129 @@
...
+          "type": "text",
+          "default": "data"
+        }
+      ]
+    },
+    "outputs_columns": {
+      "oneOf": [
+        {
+          "error_msg": "Need to specify the name(s) of the key(s) that defines the output columns of the data.",
+          "description": "You must specify which columns will be the output according to the task. By default the last column is selected.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]: "Output column(s) of the dataset. By default, the last column is selected as output."

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 5]
```diff
@@ -1,118 +1,129 @@
...
+          "description": "The data has to be inside a key in your JSON files. Write the name of the key that constains the data.",
+          "type": "text",
+          "default": "data"
+        }
+      ]
+    },
+    "outputs_columns": {
+      "oneOf": [
+        {
+          "error_msg": "Need to specify the name(s) of the key(s) that defines the output columns of the data.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]: "The specified column name(s) is(are) not valid."

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 6]
```diff
@@ -1,118 +1,129 @@
...
-                        "default": 0.1,
-                        "minimum": 0,
-                        "maximum": 1
-                    }
-                ]
+    "splits_in_folders": {
+      "oneOf": [
+        {
+          "error_msg": "Need to select 'true' if you have folders that define the splits, otherwise 'false'.",
+          "description": "If your data has folders that define the splits select 'true', otherwise 'false'.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json:42]: "True allows you to indicate that the data are divided into "train", "test" and "validation" folders. False otherwise, indicates there are no previous divisions of the dataset."
```suggestion
          "description": "Option that allows you to indicate that the data are divided into "train", "test" and "validation" folders. Otherwise, it is assumed that there are no previous divisions of the dataset.",
```

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 7]
```diff
@@ -1,118 +1,129 @@
...
-                        "type": "float",
-                        "default": 0.1,
-                        "minimum": 0,
-                        "maximum": 1
-                    }
-                ]
+    "splits_in_folders": {
+      "oneOf": [
+        {
+          "error_msg": "Need to select 'true' if you have folders that define the splits, otherwise 'false'.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]: "You must specify true or false."
```suggestion
          "error_msg": "You must specify true or false.",
```

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 8]
```diff
@@ -1,118 +1,129 @@
...
+    }
+  },
+  "splits": {
+    "display": "div",
+    "properties": {
+      "train_size": {
+        "oneOf": [
+          {
+            "error_msg": "The size of the training set must be between 0 and 1.",
+            "description": "The training set contains the data to be used for training a model. Must be defined between 0 and 100% of the data.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]: My guess is that this information (what is train, validation test sets) should be in the app, not in the parameter description.
I should abbreviate this line to: 
"Percentage of the data that will be assigned to produce the training dataset (in the range 0 to 1, where 0 is 0% and 1 is 100%)."

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 9]
```diff
@@ -1,118 +1,129 @@
...
+            "minimum": 0,
+            "maximum": 1
+          }
+        ]
+      },
+      "test_size": {
+        "oneOf": [
+          {
+            "error_msg": "The test set size must be between 0 and 1.",
+            "description": "The test set contains the data that will be used to evaluate a model. Must be defined between 0 and 100% of the data.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]: Same here: 
"Percentage of the data that will be assigned to produce the test dataset (in the range 0 to 1, where 0 is 0% and 1 is 100%)."

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 10]
```diff
@@ -1,118 +1,129 @@
...
+            "default": 0.2,
+            "minimum": 0,
+            "maximum": 1
+          }
+        ]
+      },
+      "val_size": {
+        "oneOf": [
+          {
+            "error_msg": "The validation set size must be between 0 and 1.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json:79]: Percentage of the data that will be assigned to produce the validation dataset (in the range 0 to 1, where 0 is 0% and 1 is 100%).

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 11]
```diff
@@ -1,118 +1,129 @@
...
+      }
+    },
+    "more_options": {
+      "display": "modal",
+      "properties": {
+        "shuffle": {
+          "oneOf": [
+            {
+              "error_msg": "Must be true or false, choose if you want to shuffle the data when separating the sets.",
+              "description": "Determines whether the data will be shuffle when defining the sets or not. It must be true for shuffle the data, otherwise false.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]:
```suggestion
              "description": "This option indicates that the data will be randomly shuffled prior to generating the training, validation and test sets.",
```

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 12]
```diff
@@ -1,118 +1,129 @@
...
+        ]
+      }
+    },
+    "more_options": {
+      "display": "modal",
+      "properties": {
+        "shuffle": {
+          "oneOf": [
+            {
+              "error_msg": "Must be true or false, choose if you want to shuffle the data when separating the sets.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]:
```suggestion
              "error_msg": "Shuffle should be True or False.",
```

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 13]
```diff
@@ -1,118 +1,129 @@
...
-                        }
-                    ]
-                }
-            },
-            "type": "object"
+        "seed": {
+          "oneOf": [
+            {
+              "error_msg": "Must be an integer greater than or equal to 0.",
+              "description": "A seed defines a value with which the same mixture of data will always be obtained. It must be an integer greater than or equal to 0.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]:
```suggestion
              "description": "Positive integer that defines a randomness seed, which is a mechanism that allows to control randomness and make the results repeatable, even if within the same process there are random steps.".,
```

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 14]
```diff
@@ -1,118 +1,129 @@
...
+              "default": 0,
+              "minimum": 0
+            }
+          ]
         },
-        "type": "object"
+        "stratify": {
+          "oneOf": [
+            {
+              "error_msg": "Must be true or false, choose if you want to separate the data into sets with the same proportion of samples per class as the original set or not.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]:
```suggestion
              "error_msg": "Must be true or false.",
```

File: DashAI/back/dataloaders/params_schemas/jsondataloader.json [hunk 15]
```diff
@@ -1,118 +1,129 @@
...
+              "description": "Defines whether the data will be proportionally separated according to the distribution of classes in each set.",
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/params_schemas/jsondataloader.json]: "This option ensures that when dividing the dataset into train, test, and validation sets, the original distribution of labels in the input dataset will be preserved in each of the resulting datasets. This option is applicable only to classification problems"

**Review by pbadillatorrealba — CHANGES_REQUESTED** [review]
I simplified some messages in order to make them more consistent and easy to understand with the things that a common DashAI will see.
@maximi4321 , please review them and check if they are correct.

**Comment by maximi4321** [comment]
The latest commit modifies the messages.

I think this improves the messages, makes them more concise and better explained and also considers the context in which the user would be when viewing them.

**Review by cristian-tamblay — APPROVED** [review]
